python/ray/util/queue.py

```diff
@@ -212,8 +222,28 @@ def get_nowait_batch(self, num_items: int) -> List[Any]:
 
         return ray.get(self.actor.get_nowait_batch.remote(num_items))
 
+    def shutdown(self, force: bool = False) -> None:
+        """Terminates the underlying QueueActor.
+
+        All of the resources reserved by the queue will be released.
+
+        Args:
+            force (bool): If True, forcefully kill the actor, causing an
+            immediate failure. If False, graceful
+            actor termination will be attempted first, before falling back
+            to a forceful kill.
+        """
+        if self.actor:
+            if force:
+                ray.kill(self.actor, no_restart=True)
+            else:
+                try:
+                    ray.get(self.actor.__ray_terminate__.remote())
+                except RayActorError:
```

Comment:
I'm pretty sure this will *always* raise a `RayActorError`. Ideally, you would do something like the following:
```python
done_ref = self.actor.__ray_terminate__.remote()
done, not_done = ray.wait([done_ref], timeout=GRACE_PERIOD)
if not_done:
    ray.kill(self.actor, no_restart=True)
```

Reply:
not_done codepath will be triggered if the actor didn't successfully exit within the grace period